File: src/test/java/com/github/kayjamlang/tests/expressions/ReturnExpressionTest.java
package com.github.kayjamlang.tests.expressions;

import com.github.kayjamlang.core.expressions.Expression;
import com.github.kayjamlang.core.expressions.ReturnExpression;
import com.github.kayjamlang.core.expressions.ValueExpression;
import com.github.kayjamlang.tests.TestsUtils;
import org.junit.Test;

import static org.junit.Assert.*;

public class ReturnExpressionTest {

    @Test
    public void test() throws Exception {
        Expression expression = TestsUtils.parse("return 2005");

        assertNotNull(expression);
        assertSame(ReturnExpression.class, expression.getClass());

        ReturnExpression returnExpression = (ReturnExpression) expression;
        assertSame(ValueExpression.class, returnExpression.expression.getClass());

        ValueExpression constant = (ValueExpression) returnExpression.expression;
        assertEquals(2005, constant.value);
    }
}

File: src/test/scala/com/github/kayjamlang/tests/expressions/ReturnExpressionTest.scala
package com.github.kayjamlang.tests.expressions

import com.github.kayjamlang.core.{KayJamLexer, KayJamParser}
import com.github.kayjamlang.core.expressions.{ReturnExpression, ValueExpression}
import org.junit.Assert._
import org.junit.{BeforeClass, Test}

object ReturnExpressionTest {
  private var parser: KayJamParser = null

  @BeforeClass def prepare(): Unit = {
    parser = new KayJamParser(new KayJamLexer("return 2005"))
  }
}

class ReturnExpressionTest {
  @Test
  @throws[Exception]
  def test(): Unit = {
    val expression = ReturnExpressionTest.parser.readExpression
    assertNotNull(expression)
    assertSame(classOf[ReturnExpression], expression getClass)
    val returnExpression = expression.asInstanceOf[ReturnExpression]
    assertSame(classOf[ValueExpression], returnExpression.expression getClass)
    val constant = returnExpression.expression.asInstanceOf[ValueExpression]
    assertEquals(2005, constant.value)
  }
}